### DISCUSSION TITLE
Help, Error on Visual Studio .NET C-sharp

### DISCUSSION THREAD

**Discussion by Jamesdagless** [dotnet/runtime#64636] [Q&A] [answered]
Hi i'm having an error come up on my terminal for visual studio (just downloaded) when i try to play the code. I'm currently using a mac Version 10.13.6 any one got any ideas? Thankyou 
the terminal is reading: 

  Referenced from: /usr/local/share/dotnet/dotnet (which was built for Mac OS X 11.6)
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: ____chkstk_darwin
  Referenced from: /usr/local/share/dotnet/dotnet (which was built for Mac OS X 11.6)
  Expected in: /usr/lib/libSystem.B.dylib

And this is the Problem Report for dotnet:

Process:               dotnet [4327]
Path:                  /usr/local/share/dotnet/dotnet
Identifier:            dotnet
Version:               0
Code Type:             X86-64 (Native)
Parent Process:        vsdbg [4326]
Responsible:           dotnet [4327]
User ID:               501

Date/Time:             2022-02-01 17:22:34.536 +0000
OS Version:            Mac OS X 10.13.6 (17G14042)
Report Version:        12
Anonymous UUID:        73042E95-EBBC-D52A-CBE5-F63102341C18

Sleep/Wake UUID:       7B70B1F6-2A8A-4B5F-8DF2-B388ACC7A76F

Time Awake Since Boot: 14000 seconds
Time Since Wake:       3700 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x4] Symbol missing

Dyld Error Message:
  Symbol not found: ____chkstk_darwin
  Referenced from: /usr/local/share/dotnet/dotnet (which was built for Mac OS X 11.6)
  Expected in: /usr/lib/libSystem.B.dylib

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   dyld                          	0x000000010c306256 __abort_with_payload + 10
1   dyld                          	0x000000010c305bc8 abort_with_payload_wrapper_internal + 89
2   dyld                          	0x000000010c305bf5 abort_with_payload + 9
3   dyld                          	0x000000010c2d9482 dyld::halt(char const*) + 354
4   dyld                          	0x000000010c2d95a9 dyld::fastBindLazySymbol(ImageLoader**, unsigned long) + 170
5   libdyld.dylib                 	0x00007fff573e0292 dyld_stub_binder + 282
6   ???                           	0x0000000108851320 0 + 4437906208
7   dotnet                        	0x000000010884a9eb 0x108841000 + 39403
8   dotnet                        	0x000000010884b150 0x108841000 + 41296
9   libdyld.dylib                 	0x00007fff573e0015 start + 1

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000002000209  rbx: 0x0000000000000004  rcx: 0x00007ffee73bd748  rdx: 0x00007ffee73bdbc0
  rdi: 0x0000000000000006  rsi: 0x0000000000000004  rbp: 0x00007ffee73bd7a0  rsp: 0x00007ffee73bd748
   r8: 0x00007ffee73bd7c0   r9: 0x0000000000000000  r10: 0x0000000000000060  r11: 0x0000000000000246
  r12: 0x0000000000000060  r13: 0x00007ffee73bdbc0  r14: 0x0000000000000006  r15: 0x0000000000000000
  rip: 0x000000010c306256  rfl: 0x0000000000000246  cr2: 0x000000010c356fe0
  
Logical CPU:     0
Error Code:      0x02000209
Trap Number:     133

Binary Images:
       0x108841000 -        0x10884cff7 +dotnet (0) <91994B15-7B20-3867-AD27-3EA469C14872> /usr/local/share/dotnet/dotnet
       0x10c2d5000 -        0x10c31fadf  dyld (551.5) <CB9BFB56-4511-36F1-A546-891FF770C01C> /usr/lib/dyld
    0x7fff54b3c000 -     0x7fff54b6fff7  libclosured.dylib (551.5) <112BC241-6626-3848-8DD8-B34B5B6F7ABC> /usr/lib/closure/libclosured.dylib
    0x7fff55076000 -     0x7fff55077ffb  libSystem.B.dylib (1252.50.4) <5A02016C-8F9D-3F47-8C39-AD2CD4F9F11D> /usr/lib/libSystem.B.dylib
    0x7fff5532e000 -     0x7fff55384fff  libc++.1.dylib (400.9) <142D6292-F30F-3949-9F37-3C97A579F540> /usr/lib/libc++.1.dylib
    0x7fff55385000 -     0x7fff553a9ff7  libc++abi.dylib (400.8.2) <EF5E37D7-11D9-3530-BE45-B986612D13E2> /usr/lib/libc++abi.dylib
    0x7fff567b1000 -     0x7fff56ba23b7  libobjc.A.dylib (723) <616AB826-9179-3924-AA61-AF760363A3F5> /usr/lib/libobjc.A.dylib
    0x7fff57276000 -     0x7fff5727aff7  libcache.dylib (80) <096D3699-58D2-3D23-9512-BC88DAE7B16C> /usr/lib/system/libcache.dylib
    0x7fff5727b000 -     0x7fff57285ff3  libcommonCrypto.dylib (60118.50.1) <B4E77BD4-A809-37E1-8A79-25955E0F5D9C> /usr/lib/system/libcommonCrypto.dylib
    0x7fff57286000 -     0x7fff5728dfff  libcompiler_rt.dylib (62) <4CEED002-D00B-35F8-B563-0546A5F71A43> /usr/lib/system/libcompiler_rt.dylib
    0x7fff5728e000 -     0x7fff57297ffb  libcopyfile.dylib (146.50.5) <F7A7999F-9509-3C67-8EE9-95BEE6772676> /usr/lib/system/libcopyfile.dylib
    0x7fff57298000 -     0x7fff5731dffb  libcorecrypto.dylib (562.70.3) <9C6E5CC2-272B-3740-AEDB-9EBEC98EC7F9> /usr/lib/system/libcorecrypto.dylib
    0x7fff573a5000 -     0x7fff573deff7  libdispatch.dylib (913.60.3) <285526B6-A2A6-316D-A1CC-37803B71E203> /usr/lib/system/libdispatch.dylib
    0x7fff573df000 -     0x7fff573fcff7  libdyld.dylib (551.5) <49ABA86D-DD48-3133-9B14-B9A564EEBC66> /usr/lib/system/libdyld.dylib
    0x7fff573fd000 -     0x7fff573fdffb  libkeymgr.dylib (28) <7CC7B5E1-AB7F-3495-A26A-396461BCB66B> /usr/lib/system/libkeymgr.dylib
    0x7fff5740b000 -     0x7fff5740bff7  liblaunch.dylib (1205.70.16) <265EE555-E988-3EE5-A39E-028FB3A5FE11> /usr/lib/system/liblaunch.dylib
    0x7fff5740c000 -     0x7fff57410ffb  libmacho.dylib (906) <F5F16E2F-F961-3FF4-B02F-69BBDB57B8DC> /usr/lib/system/libmacho.dylib
    0x7fff57411000 -     0x7fff57413fff  libquarantine.dylib (86.70.1) <262CECFE-A493-3EC6-8608-CEA93757265E> /usr/lib/system/libquarantine.dylib
    0x7fff57414000 -     0x7fff57415ff3  libremovefile.dylib (45) <EA24BFB6-24B7-3841-8AAC-FAF400910706> /usr/lib/system/libremovefile.dylib
    0x7fff57416000 -     0x7fff5742dfff  libsystem_asl.dylib (356.70.1) <9481D82D-CFB7-3032-A07F-88F7F59CAC2C> /usr/lib/system/libsystem_asl.dylib
    0x7fff5742e000 -     0x7fff5742efff  libsystem_blocks.dylib (67) <FA2C3663-9DBB-3AFB-80A3-E77BE0816AD0> /usr/lib/system/libsystem_blocks.dylib
    0x7fff5742f000 -     0x7fff574b8ff7  libsystem_c.dylib (1244.50.9) <25DD83D8-80CA-3DFF-8626-FE704911F19C> /usr/lib/system/libsystem_c.dylib
    0x7fff574b9000 -     0x7fff574bcffb  libsystem_configuration.dylib (963.50.9) <61B26E1D-FEFF-38C6-AE0E-A6DD90145EB2> /usr/lib/system/libsystem_configuration.dylib
    0x7fff574bd000 -     0x7fff574c0ffb  libsystem_coreservices.dylib (51) <E8FA663F-1B0F-3B12-8F97-31A46E6B4D68> /usr/lib/system/libsystem_coreservices.dylib
    0x7fff574c1000 -     0x7fff574c2fff  libsystem_darwin.dylib (1244.50.9) <6195EEA1-83A7-35F0-8147-5AB031020CD0> /usr/lib/system/libsystem_darwin.dylib
    0x7fff574c3000 -     0x7fff574c9ffb  libsystem_dnssd.dylib (878.70.4) <0D2C9583-5D28-375B-AEB4-F66F18C78322> /usr/lib/system/libsystem_dnssd.dylib
    0x7fff574ca000 -     0x7fff57513ff7  libsystem_info.dylib (517.30.1) <9C2C96F3-638B-34E7-A2BD-24C643EAF957> /usr/lib/system/libsystem_info.dylib
    0x7fff57514000 -     0x7fff5753aff7  libsystem_kernel.dylib (4570.71.82.8) <C34BA704-FAFF-3DDB-9827-5930B5BEF134> /usr/lib/system/libsystem_kernel.dylib
    0x7fff5753b000 -     0x7fff57586fcb  libsystem_m.dylib (3147.50.1) <B3172533-77D2-3416-8487-4A505198E9E1> /usr/lib/system/libsystem_m.dylib
    0x7fff57587000 -     0x7fff575a6ff7  libsystem_malloc.dylib (140.50.7) <821E16C8-7A7E-3269-9167-10F4F413BF93> /usr/lib/system/libsystem_malloc.dylib
    0x7fff575a7000 -     0x7fff576d7ff7  libsystem_network.dylib (1229.70.2) <07BFD80A-E7EB-3DEB-B765-5E1659453114> /usr/lib/system/libsystem_network.dylib
    0x7fff576d8000 -     0x7fff576e2ffb  libsystem_networkextension.dylib (767.70.2) <A90BAFFD-AC3D-3078-90B2-71463DBFD4A5> /usr/lib/system/libsystem_networkextension.dylib
    0x7fff576e3000 -     0x7fff576ecff3  libsystem_notify.dylib (172) <738B3BB0-4F20-3446-A3B6-095D796E5903> /usr/lib/system/libsystem_notify.dylib
    0x7fff576ed000 -     0x7fff576f4ff7  libsystem_platform.dylib (161.50.1) <90A2A62F-8690-3FA7-A4D0-9085BF2FA44A> /usr/lib/system/libsystem_platform.dylib
    0x7fff576f5000 -     0x7fff57700fff  libsystem_pthread.dylib (301.50.1) <283E64A7-A2B2-3212-95BA-4D21F9AE36CF> /usr/lib/system/libsystem_pthread.dylib
    0x7fff57701000 -     0x7fff57704fff  libsystem_sandbox.dylib (765.70.2) <C29E1B7A-F39B-344E-98F7-8FB32B10AE0A> /usr/lib/system/libsystem_sandbox.dylib
    0x7fff57705000 -     0x7fff57706ff3  libsystem_secinit.dylib (30) <4E84DF62-57B5-36F9-819F-3C1E7FEDA856> /usr/lib/system/libsystem_secinit.dylib
    0x7fff57707000 -     0x7fff5770eff7  libsystem_symptoms.dylib (820.60.3) <A45B4D84-2C69-3F19-83C2-2F1EEA73A7B5> /usr/lib/system/libsystem_symptoms.dylib
    0x7fff5770f000 -     0x7fff57723ff7  libsystem_trace.dylib (829.70.2) <4C4E5351-E370-3226-9B30-61E250DE3B2E> /usr/lib/system/libsystem_trace.dylib
    0x7fff57725000 -     0x7fff5772aff7  libunwind.dylib (35.3) <BAE12DA3-CF33-32AD-86B0-CCD6705ADFAA> /usr/lib/system/libunwind.dylib
    0x7fff5772b000 -     0x7fff57758fff  libxpc.dylib (1205.70.16) <26D99819-B813-320E-9E8F-9D57AD2DA63B> /usr/lib/system/libxpc.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 9180
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=202.6M resident=0K(0%) swapped_out_or_unallocated=202.6M(100%)
Writable regions: Total=17.2M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=17.2M(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Kernel Alloc Once                    8K        2 
MALLOC                            9260K        8 
MALLOC guard page                   16K        5 
STACK GUARD                       56.0M        2 
Stack                             8192K        2 
__DATA                            2336K       42 
__DATA_CONST                        16K        2 
__LINKEDIT                       193.4M        4 
__TEXT                            9508K       42 
shared memory                        8K        3 
===========                     =======  ======= 
TOTAL                            278.0M      102

**Comment by vcsjones** [comment, marked as answer]
It suspect this is because your version of macOS is too old to run .NET 6. The [release notes](https://github.com/dotnet/core/blob/main/release-notes/6.0/supported-os.md) indicate that 10.15 is the minimum OS requirements to run .NET 6.

.NET 6 is built with 10.14 as a minimum OS requirement, so the build is likely using something that is not available on a 10.13 system.

https://github.com/dotnet/runtime/blob/bbfc15ca692c32a60dacee5e204e638418e5d0bd/eng/native/configurecompiler.cmake#L423
